src/psyclone/psyGen.py

```diff
@@ -4490,6 +4506,252 @@ def nodes_to_code_block(parent, statements):
+                 represents a scalar.
+        :rtype: list
+        '''
+        from fparser.two.utils import walk_ast
+        from fparser.two import Fortran2003
+        shape = []
+        for dim in walk_ast(dimensions.items, [Fortran2003.Assumed_Shape_Spec,
+                                               Fortran2003.Explicit_Shape_Spec,
+                                               Fortran2003.Assumed_Size_Spec]):
+            if isinstance(dim, Fortran2003.Assumed_Size_Spec):
+                raise NotImplementedError(
+                    "Could not process {0}. Assumed-size arrays"
+                    " are not supported.".format(dimensions))
+            elif isinstance(dim, Fortran2003.Assumed_Shape_Spec):
+                shape.append(None)
+            elif isinstance(dim, Fortran2003.Explicit_Shape_Spec):
+                if isinstance(dim.items[1],
+                              Fortran2003.Int_Literal_Constant):
+                    shape.append(int(dim.items[1].items[0]))
+                else:
+                    raise NotImplementedError(
+                        "Could not process {0}. Only integer "
+                        "literals are supported for explicit shape"
+                        " array declarations.".format(dimensions))
+            else:
+                raise InternalError(
+                    "Reached end of loop body and {0} has"
+                    " not been handled.".format(type(dim)))
+        return shape
+
+    def process_declarations(self, parent, nodes, arg_list):
+        '''
+        Transform the variable declarations in fparser2 parse tree into
+        symbols in the PSyIR parent node symbol table.
+
+        :param parent: PSyIR node in which to insert the symbols found.
+        :type parent: :py:class:`psyclone.psyGen.KernelSchedule`
+        :param nodes: fparser2 AST nodes to search for declaration statements.
+        :type nodes: list of :py:class:`fparser.two.utils.Base`
+        :param arg_list: fparser2 AST node containing the argument list.
+        :type arg_list: :py:class:`fparser.Fortran2003.Dummy_Arg_List`
+        :raises NotImplementedError: The provided declarations contain
+                                     attributes which are not supported yet.
+        '''
+        from fparser.two.utils import walk_ast
+        from fparser.two import Fortran2003
+
+        def iterateitems(nodes):
+            '''
+            At the moment fparser nodes can be of type None, a single element
+            or a list of elements. This helper function provide a common
+            iteration interface. This could be improved when fpaser/#170 is
+            fixed.
+            :param nodes: fparser2 AST node.
+            :type nodes: None or List or :py:class:`fparser.two.utils.Base`
+            :return: Returns nodes but always encapsulated in a list
+            :rtype: list
+            '''
+            if nodes is None:
+                return []
+            else:
```

Comment:
pylint points out that this `else` is unnecessary because it follows a `return`

Reply:
Fixed